internal/database/repo_update_jobs.go

```diff
@@ -0,0 +1,180 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/lib/pq"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/executor"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+)
+
+type RepoUpdateJobStore interface {
+	Handle() *basestore.Store
+	Create(ctx context.Context, opts RepoUpdateJobOpts) (types.RepoUpdateJob, bool, error)
+	List(ctx context.Context, opts ListRepoUpdateJobOpts) ([]types.RepoUpdateJob, error)
+	SaveUpdateJobResults(ctx context.Context, jobID int, opts SaveUpdateJobResultsOpts) error
```

Comment:
I think the `UpdateJob` is implicit in the `RepoUpdateJobStore` and `SaveUpdate` -- two verbs in sequence -- sounds odd.

```suggestion
	SetResults(ctx context.Context, jobID int, opts SaveUpdateJobResultsOpts) error
```